heartbeat/monitor/heartbeatMessage.go

```diff
@@ -0,0 +1,62 @@
+package monitor
+
+import (
+	"sort"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go/heartbeat/data"
+)
+
+type heartbeatMessages struct {
+	activePids         []core.PeerID
+	activeHeartbeats   map[core.PeerID]*data.PubKeyHeartbeat
+	inactiveHeartbeats map[core.PeerID]*data.PubKeyHeartbeat
+}
+
+func newHeartbeatMessages() *heartbeatMessages {
+	return &heartbeatMessages{
+		activePids:         make([]core.PeerID, 0),
+		activeHeartbeats:   make(map[core.PeerID]*data.PubKeyHeartbeat),
+		inactiveHeartbeats: make(map[core.PeerID]*data.PubKeyHeartbeat),
+	}
+}
+
+func (instance *heartbeatMessages) addMessage(pid core.PeerID, message *data.PubKeyHeartbeat) {
+	if message.IsActive {
+		instance.activePids = append(instance.activePids, pid)
+		instance.activeHeartbeats[pid] = message
```

Comment:
```suggestion
                if (pid < instance.activePid) {
		    instance.activePid = pid
		    instance.activeHeartbeat = message
		}
```
this way you just store the message for the lowerst peerID, so you dont have to sort to find it after that

Reply:
refactored, now the `heartbeatMessages` struct does not contain any maps. Removed several for... loops in the process. 
Good suggestion :+1: